File: src/main/java/com/heekng/celloct/repository/WorkRepository.java
package com.heekng.celloct.repository;

import com.heekng.celloct.entity.Work;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.List;

public interface WorkRepository extends JpaRepository<Work, Long> {
    List<Work> findByWorkTimeWorkDateAndStaffId(LocalDate workDate, Long staffId);

    List<Work> findByWorkTimeWorkDateBetweenAndStaffId(LocalDate start, LocalDate end, Long staffId);
}

File: src/main/kotlin/com/heekng/celloct/repository/WorkRepository.kt
package com.heekng.celloct.repository

import com.heekng.celloct.entity.Work
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDate

interface WorkRepository : JpaRepository<Work, Long> {

    fun findByWorkTimeWorkDateAndStaffId(workDate: LocalDate, staffId: Long): List<Work>

    fun findByWorkTimeWorkDateBetweenAndStaffId(start: LocalDate, end: LocalDate, staffId: Long): List<Work>

    fun findByIdAndStaffId(id: Long, staffId: Long): Work?

}